org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/DeviceRunner.java

```diff
@@ -63,31 +62,29 @@
 	private long calculateTimeout(Collection<IRunnableDevice<?>> devices) {
 		long time = Long.MIN_VALUE;
 		for (IRunnableDevice<?> device : devices) {
-			if (device instanceof AbstractRunnableDevice) {
-				Object model = ((AbstractRunnableDevice<?>)device).getModel();
-				if (model instanceof IMalcolmModel) {
-					time = Long.MAX_VALUE; // no timeout for malcolm scans
-					break;
-					// TODO: use estimated scan time (x2?)
-				}
-				long timeout = -1;
-				if (model instanceof ITimeoutable) {
-					timeout = ((ITimeoutable)model).getTimeout();
-				    if (timeout<0 && model instanceof IDetectorModel) {
-				    	IDetectorModel dmodel = (IDetectorModel)model;
-				    	timeout = Math.round(dmodel.getExposureTime());
-				    }
-				} else if (model instanceof IDetectorModel) {
-			    	IDetectorModel dmodel = (IDetectorModel)model;
-			    	timeout = (long)Math.ceil(dmodel.getExposureTime());
-				}
-				time = Math.max(time, timeout);
-			}
+			time = Math.max(time, getTimeout(device));
 		}
 		if (time<=0) time = 10; // seconds
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)